File: moi/src/main/java/com/moi/model/GoldEntry.java
package com.moi.model;

import javax.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "gold_entries")
public class GoldEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "gold_details", nullable = false)
    private String goldDetails;

    @Column(name = "entry_date")
    private LocalDateTime entryDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "event_id", nullable = false)
    private Event event;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "contributor_id", nullable = false)
    private Contributor contributor;
}

File: moi-dotnet/MoiBackend.Core/Entities/GoldEntry.cs
namespace MoiBackend.Core.Entities;

public class GoldEntry
{
    public long Id { get; set; }
    public string GoldDetails { get; set; } = string.Empty;
    public DateTime EntryDate { get; set; }

    public long EventId { get; set; }
    public Event Event { get; set; } = null!;

    public long ContributorId { get; set; }
    public Contributor Contributor { get; set; } = null!;
}
